openmmforcefields/generators/template_generators.py

```diff
@@ -730,6 +743,20 @@ def generate_residue_template(self, molecule, residue_atoms=None):
         kwargs = {}
         if "write_unused" in signature(params.write).parameters:
             kwargs["write_unused"] = True
+        # import pdb; pdb.set_trace()
+        # TODO MMH
+        # get the types
+        # check if they are in our cache
+        # remove and add them as needed
+        # see if this fixes it
+        # params.atom_types.keys()
+        for atom_type in params.atom_types.copy().keys():
+            if atom_type not in self._gaff_atom_type_cache:
+                self._gaff_atom_type_cache.add(atom_type)
+                print(f"added {atom_type} to cache")
+            else:
+                del params.atom_types[atom_type]
+                print(f"del {atom_type}")
```

Comment:
I will clean this up and explain it with doc strings, I just wanted to get this tested on CI to make sure it worked (not just locally)

but the tl;dr is we need this or else we end up trying to register duplicate atom types with openmm as we loop over molecules and generate parameters

Reply:
btw, we didn't need with the old way of doing things since older versions of ambertools/parmchk2 didn't print out atom types that existed in `gaff.dat` with the way we ran parmchk2 (new versions of parmchk2 started adding atom type `c5` to the frcmod file)